cdap-api/src/main/java/co/cask/cdap/api/workflow/WorkflowConditionNode.java

```diff
@@ -42,7 +59,25 @@ public WorkflowConditionNode(String nodeId, String predicateClassName, List<Work
     return elseBranch;
   }
 
+  @Nullable
   public String getPredicateClassName() {
     return predicateClassName;
   }
+
+  @Nullable
```

Comment:
Similar to above, this shouldn't be null. If the condition comes from a `Predicate` class, we should be able to instantiate a spec that represents that.